src/python/tools/preprocess.py

```diff
@@ -545,8 +569,12 @@ def parse_args(config_dict, args):
     arg_dict = vars(args)
     config_dict = update_param(config_dict, arg_dict)
     set_up_files(args.output_directory)
-
-    config_dict.update({"dataset": arg_dict.get("dataset")})
+    
+    if arg_dict.get("dataset") is None:
+        config_dict.update({"dataset": "custom"})
```

Comment:
I might have missed it if you already did, but add a test for this "custom" case